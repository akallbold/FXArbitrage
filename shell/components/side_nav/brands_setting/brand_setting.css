@import '../../../../../node_modules/playa-styleguide/src/stylesheets/variables.css';

.statusWrapper {
  display: flex;
  align-self: center;
}
.status {
  height: 10px;
  width: 10px;
  border-radius: 50%;
  background-color: var(--green);
}

.brandItem {
  display: flex;
  height: 15px; 
  width: 222px;
  padding: 5px;
  margin-top: 15px;
}

.brandContent {
  display: flex;
  align-self: center;
}

.brandItemInactive {
  background-color: var(--white);
  padding-left: 7px;
}

.brandItemActive {
  background-color: var(--hyfn-blue-light-fill);
  border-bottom: 1px solid var(--hyfn-blue-light-border);
  border-left: 3px solid var(--hyfn-blue);
}

.brandItemInactive ~ .brandItemActive {
  border-bottom: 1px solid var(--hyfn-blue-light-border);
}

.icon {
  margin: 0 13px 0 17px;
}

.icon svg {
  width: 15px;
  height: 15px;
}

.brandItemActive .icon svg {
  fill: var(--hyfn-blue);
}

.brandItemInactive .icon svg {
  fill: var(--text-grey-light);
}

.brands {
  display: flex;
  align-self: center;
  padding-left: 20px;
  height: 14px;
  width: 54px;
  color: var(--text-grey-light);
  font-family: var(--font-family-regular);
  font-size: 12px;
  font-weight: 600;
  line-height: 14px;
}

.name {
  height: 17px;
  width: 132px;
  color: var(--text-grey-dark);
  font-family: var(--font-family-regular);
  font-size: 12px;
  font-weight: 600;
  line-height: 17px;
  text-transform: capitalize;
}
