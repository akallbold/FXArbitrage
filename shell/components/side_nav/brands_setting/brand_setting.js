import React from 'react'
// import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import {
  AccountInfoIcon,
  AdPixelsIcon,
  ConnectedPropertiesIcon
} from 'playa-styleguide'
import Styles from './brand_setting.css'

export const BRAND_SETTING_NAMES = ['account info', 'ad pixel', 'connected properties']

const renderIcon = (brandName) => {
  switch (brandName) {
    case 'account info':
      return <AccountInfoIcon />
    case 'ad pixel':
      return <AdPixelsIcon />
    case 'connected properties':
      return <ConnectedPropertiesIcon />
    default:
      return null
  }
}

const rowClassNameHelper = (active) => {
  const activeClass = active ? Styles.brandItemActive : Styles.brandItemInactive
  return `${Styles.brandItem} ${activeClass}`
}

export const BrandSettingStatusItem = ({ name, active }) => (
  <div className={rowClassNameHelper(active)} data-active={active}>
    <div className={Styles.brandContent}>
      <div className={Styles.icon}>{renderIcon(name)}</div>
      <div className={Styles.name} data-field='name'>{name}</div>
    </div>
  </div>
)

export const BrandSetting = () => {
  const BrandRows = BRAND_SETTING_NAMES.map(name => (
    <BrandSettingStatusItem
      key={name}
      name={name}
    />
  ))

  return (
    <div>
      <div className={Styles.brandItem}>
        <div className={Styles.brands}>Brand</div>
      </div>
      { BrandRows }
    </div>
  )
}

// We will shape this later. Still debating what this will look like.

// brandStatus.propTypes = {
//   BrandSettingStatuses: PropTypes.shape({
//   })
// }

// brandStatusItem.propTypes = {
// }

// Will update this when i hook up actual values from store.
// TODO: hook this up once i define what data looks like coming back.
const mapStateToProps = () => ({
  // BrandSettingStatuses: state.platforms
})

export default connect(mapStateToProps)(BrandSetting)
