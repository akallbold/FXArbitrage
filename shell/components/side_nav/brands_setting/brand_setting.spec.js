import React from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { BrandSetting, BrandSettingStatusItem, BRAND_SETTING_NAMES } from './brand_setting'

describe('BRAND_SETTING_NAMES', () => {
  it('should have the platform names in the correct order', () => {
    expect(BRAND_SETTING_NAMES).toEqual(['account info', 'ad pixel', 'connected properties'])
  })
})

describe('<BrandSetting />', () => {
  it('should render properly given props', () => {
    const component = mount(
      <BrandSetting BrandSettingStatuses={['account info', 'ad pixel', 'connected properties']} />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('<BrandSettingStatusItem />', () => {
  it('should render properly given props', () => {
    const component = shallow(
      <BrandSettingStatusItem name='account info' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
