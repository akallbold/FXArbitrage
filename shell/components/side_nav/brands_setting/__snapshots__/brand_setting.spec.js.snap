// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandSetting /> should render properly given props 1`] = `
<BrandSetting
  BrandSettingStatuses={
    Array [
      "account info",
      "ad pixel",
      "connected properties",
    ]
  }
>
  <div>
    <div
      className="brandItem"
    >
      <div
        className="brands"
      >
        Brand
      </div>
    </div>
    <BrandSettingStatusItem
      name="account info"
    >
      <div
        className="brandItem brandItemInactive"
      >
        <div
          className="brandContent"
        >
          <div
            className="icon"
          >
            <test-file-stub />
          </div>
          <div
            className="name"
            data-field="name"
          >
            account info
          </div>
        </div>
      </div>
    </BrandSettingStatusItem>
    <BrandSettingStatusItem
      name="ad pixel"
    >
      <div
        className="brandItem brandItemInactive"
      >
        <div
          className="brandContent"
        >
          <div
            className="icon"
          >
            <test-file-stub />
          </div>
          <div
            className="name"
            data-field="name"
          >
            ad pixel
          </div>
        </div>
      </div>
    </BrandSettingStatusItem>
    <BrandSettingStatusItem
      name="connected properties"
    >
      <div
        className="brandItem brandItemInactive"
      >
        <div
          className="brandContent"
        >
          <div
            className="icon"
          >
            <test-file-stub />
          </div>
          <div
            className="name"
            data-field="name"
          >
            connected properties
          </div>
        </div>
      </div>
    </BrandSettingStatusItem>
  </div>
</BrandSetting>
`;

exports[`<BrandSettingStatusItem /> should render properly given props 1`] = `
<div
  className="brandItem brandItemInactive"
>
  <div
    className="brandContent"
  >
    <div
      className="icon"
    >
      <test-file-stub />
    </div>
    <div
      className="name"
      data-field="name"
    >
      account info
    </div>
  </div>
</div>
`;
